python/lsst/ts/idl/enums/TunableLaser.py

```diff
@@ -0,0 +1,30 @@
+import enum
+
+
+class LaserDetailedState(enum.IntEnum):
+    """An enumeration class for handling the TunableLaser's substates.
+
+    These enumerations listed here correspond to the ones found in the
+    detailedState enum located in ts_xml under the TunableLaser folder within
+    the TunableLaser_Events.xml.
+
+    Attributes
+    ----------
+
+    NONPROPAGATING : `int`
+        Corresponds to the nonpropgating state.
+    PROPAGATINGSTATE : `int`
+        Corresponds to the propagating state.
+
+    """
+    NONPROPAGATING = 1
+    PROPAGATING = 2
+
+
+class LaserErrorCode(enum.IntEnum):
+    """Laser error codes
+    """
+    ASCII_ERROR = 7301
+    GENERAL_ERROR = 7302
+    TIMEOUT_ERROR = 7303
+    HW_CPU_ERROR = 7304
```

Comment:
I'm a bit uneasy about this. On the plus side it means users don't need the package to understand the error codes, and that's definitely a nice feature. On the minus side it makes it harder to add new code and these constants aren't in the XML.

I wonder if users will need these codes? When a CSC goes into fault it publishes both a code and an explanation, and I suspect most users will care mostly about the  latter.

Reply:
I think it was the under a system where each subsystem has a range of error codes and so I decided to codify it by this methodology.

Reply:
For the record: each CSC can now have any error codes it wants. Not that I am suggesting you change these, merely that if you write a new CSC you can start at 1 or any value you think best.